core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -933,19 +925,12 @@ class TimeSeriesShard(val dataset: Dataset,
                                      partFlushChunks: Iterator[ChunkSet],
                                      timeBucket: Int) = {
     // Synchronize for safe read-modify-write behavior.
-    activelyIngesting.synchronized {
-      if (partFlushChunks.isEmpty && activelyIngesting.get(p.partID)) {
-        var endTime = p.timestampOfLatestSample
-        if (endTime == -1) endTime = System.currentTimeMillis() // this can happen if no sample after reboot
-        updatePartEndTimeInIndex(p, endTime)
-        timeBucketBitmaps.get(timeBucket).set(p.partID)
-        activelyIngesting.clear(p.partID)
-      } else if (partFlushChunks.nonEmpty && !activelyIngesting.get(p.partID)) {
-        // Partition started re-ingesting.
-        updatePartEndTimeInIndex(p, Long.MaxValue)
-        timeBucketBitmaps.get(timeBucket).set(p.partID)
-        activelyIngesting.set(p.partID)
-      }
+    if (partFlushChunks.isEmpty && activelyIngesting.get(p.partID)) {
+      var endTime = p.timestampOfLatestSample
+      if (endTime == -1) endTime = System.currentTimeMillis() // this can happen if no sample after reboot
+      updatePartEndTimeInIndex(p, endTime)
+      timeBucketBitmaps.get(timeBucket).set(p.partID)
+      activelyIngesting.clear(p.partID)
```

Comment:
Lock should probably held over the get and clear calls together. Perhaps a combined get/clear method? remove?